File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP1-2/src/constructeurAuto/Stock.java
package constructeurAuto;

import Usines.Usine;

public class Stock {
	private String idStock;
	private static int IDint = 0;
	private static char IDchar = 'A';
	private Usine usine;
	
	public Stock(Usine usine) {
		this.idStock = upID();
		this.usine = usine;
	}
	
	public Stock(Stock st) {
		this.idStock = upID();
		this.usine = st.usine;
	}

	private String upID() {
		String id = ""+ IDchar + IDint;
		if(Stock.IDint == 9999999) {
			Stock.IDint = 0;
			if(IDchar >= 'A' && IDchar <= 'Z') {
				Stock.IDchar++;
			}
			else if(IDchar >= 'a' && IDchar <= 'z') {
				Stock.IDchar++;
			}
			else if(IDchar == 'Z') {
				Stock.IDchar = 'a';
			}
			else if(IDchar == 'z') {
				Stock.IDchar = 'A';
			}
			else {
				Stock.IDchar = '@';
			}
		}
		Stock.IDint ++;
		return id;
	}

	public String getIdStock() {
		return idStock;
	}

	public Usine getUsine() {
		return usine;
	}

	public void setParking(Usine usine) {
		this.usine = usine;
	}

	@Override
	public String toString() {
		return "Stock [idStock=" + idStock + ", Usine=" + usine.toString() + "]";
	}
}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2_sofiane/Stock.php
<?php
class Stock{
    
    // attributs

    private $idStock;
    private static $IDint = 0;
    private static $IDchar = 'A';
    private $usine;

    // constructeur par initialisation

    public function __construct($usine){
    	$this->idStock = $this->upID();
        $this->usine = $usine;
    }

    private function upID(){
        $id = is_numeric("") + is_numeric(self::$IDchar) + is_numeric(self::$IDint);
        if(Stock::$IDint == 9999999){
            $stock->IDint = 0;
            if($IDchar >= 'A' && $IDchar <= 'Z'){
                $stock->IDchar++;
            }else if($IDchar >='a' && $IDchar <= 'z'){
                $stock->IDchar++;
            }else if($IDchar == 'Z'){
                $stock->IDchar = 'a';
            }else if($IDchar == 'z'){
                $stock->IDchar = 'A';
            }else{
                $stock->IDchar = '@';
            }
        }
        Stock::$IDint++;
        return $id;
    }

    // getters

    public function getIdStock(){
        return $this->idStock;
    }

    public function getUsine(){
        return $this->usine;
    }

    // setters

    public function setParking($usine){
        $this->usine = $usine;
    }

    // methode toString

    public function __toString(){
        return "Stock [idStock=" . $this->idStock . ", Usine=" . $this->usine->__toString() . "]";
    }
}